### PR TITLE
Feature/34181 - Update minicart

### PR DESCRIPTION
**Pull request by kjugi**
+ Add `popup` element
+ Add `mincart-content` element
+ Update `minicart-trigger` element
+ Update `header` module
+ Adjust `side-menu-trigger`
+ Change attribute name in `image` component
+ Add all icons to icon config

https://dev.snowdog.pro/issues/34181

### CODE REVIEW

File: components/02-elements/minicart-content/_minicart-content.scss [hunk 1]
```diff
@@ -0,0 +1,117 @@
+$minicart-content-margin      : 15px !default;
```
Comment (Igloczek) [inline components/02-elements/minicart-content/_minicart-content.scss]: Can we move this stuff to `modules`? Will be nice to at least keep the rule of not inheriting elements in elements, then we can remove it from modules.

File: components/02-elements/popup/popup.hbs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div class="popup {{ class }}">
+    {{ render (component content) contentContext }}
+</div>
+
+{{#if script }}
+<script src="{{static 'popup.js' }}"></script>
```
Comment (Igloczek) [inline components/02-elements/popup/popup.hbs]: ➡️

File: components/02-elements/minicart-content/_minicart-content.scss (hunk 1 above)
Comment (Igloczek) [reply components/02-elements/minicart-content/_minicart-content.scss]: Comment to this specific line - it's not just a `margin`, more like a `spacing`, b/c it's in use in many cases.

File: components/02-elements/popup/popup.hbs (hunk 2 above)
Comment (kjugi) [reply components/02-elements/popup/popup.hbs]: space added 🆗

File: components/02-elements/minicart-content/_minicart-content.scss (hunk 1 above)
Comment (kjugi) [reply components/02-elements/minicart-content/_minicart-content.scss]: `minicart-content` moved to the modules BUT. When we are keeping the rule of elements and modules. Popup element should be in modules too. Than we can't keep minicart-content here b/c it's using on header in popup. And header can't use popup b/c it's module too 🤔

**Comment by kjugi** [comment]
There are only style changes in the preview. The update of popup script is mentioned in [this issue](https://github.com/SnowdogApps/magento2-alpaca-ui-components/issues/68) and related to [this issue too](https://github.com/SnowdogApps/magento2-alpaca-ui-components/issues/69)

File: components/02-elements/minicart-content/_minicart-content.scss (hunk 1 above)
Comment (Igloczek) [reply components/02-elements/minicart-content/_minicart-content.scss]: I'd suggest changing the rules a little bit and make an inheritance of modules "legal", but the question is if it's technically possible and will not lead to some weird scenarios.

File: components/02-elements/popup/popup.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+(function() {
+
+  'use strict';
+
+  const popupsButtons = [...document.querySelectorAll('.popup__button')];
+
+  popupsButtons.forEach(popupButton => {
+    class Popup {
```
Comment (Igloczek) [inline components/02-elements/popup/popup.js]: The class should be outside the loop and you should just create new instances inside.

File: components/03-modules/minicart-content/_minicart-content.scss [hunk 4]
```diff
@@ -0,0 +1,116 @@
+$minicart-content-spacing     : 15px !default;
+
+$minicart-content-product-font: $font-family-sans-serif !default;
```
Comment (Igloczek) [inline components/03-modules/minicart-content/_minicart-content.scss]: Isn't it the default font?

File: components/03-modules/minicart-content/_minicart-content.scss [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        &:last-child {
+            margin-bottom: 0;
+        }
+    }
+
+    &__icon {
+        width: 20px;
+        height: 20px;
+        fill: $gray;
```
Comment (Igloczek) [inline components/03-modules/minicart-content/_minicart-content.scss]: Define variable in component scope.

Comment (Igloczek) [reply components/03-modules/minicart-content/_minicart-content.scss]: Basically most of the stuff should have a variable, to simplify customization.

Comment (kjugi) [reply components/03-modules/minicart-content/_minicart-content.scss]: variables added

File: components/03-modules/minicart-content/_minicart-content.scss (hunk 4 above)
Comment (kjugi) [reply components/03-modules/minicart-content/_minicart-content.scss]: it is, deleted

File: components/02-elements/minicart-content/_minicart-content.scss (hunk 1 above)
Comment (kjugi) [reply components/02-elements/minicart-content/_minicart-content.scss]: which weird scenarios do you mean? I moved popup to modules too and seems to working fine in header

File: components/02-elements/popup/popup.js (hunk 3 above)
Comment (kjugi) [reply components/02-elements/popup/popup.js]: moved outside the loop

Comment (kjugi) [reply components/02-elements/popup/popup.js]: tested on two same popups in header and working fine

File: components/02-elements/minicart-content/_minicart-content.scss (hunk 1 above)
Comment (kjugi) [reply components/02-elements/minicart-content/_minicart-content.scss]: tested on 2 same popups in header and working fine

**Pull request merged**